kinder/pkg/cri/nodes/containerd/alterhelper.go

```diff
@@ -48,8 +49,32 @@ func GetAlterContainerArgs() ([]string, []string) {
 
 // SetupRuntime setups the runtime
 func SetupRuntime(bc *bits.BuildContext) error {
-	// This used to hold config.toml modifications for the systemd cgroup driver,
-	// but newer kind base images have it baked in.
+	// Rewrite the containerd config (/etc/containerd/config.toml)
+	// to use the sandbox image recommended by kubeadm
+	binaryPath := "/kind/bin/kubeadm"
+	cmd := fmt.Sprintf(
+		`%[1]s config images list --kubernetes-version=$(%[1]s version -o short) 2> /dev/null | grep pause`,
+		binaryPath,
+	)
+	images, err := bc.CombinedOutputLinesInContainer("bash", "-c", cmd)
+	if err != nil {
+		return errors.Wrapf(err, "failed to execute command %q, output %v", cmd, images)
+	}
+	if len(images) != 1 {
+		return errors.Errorf("expected the output of command %q to have 1 line, got: %v", cmd, images)
+	}
+	if len(images[0]) > 0 {
+		cmd = fmt.Sprintf(
+			`sed -i 's#sandbox_image = ".*"#sandbox_image = "%s"#g' /etc/containerd/config.toml`,
```

Comment:
is the sandbox image line always present?

Reply:
Updated